pkg/gardenlet/features/features.go

```diff
@@ -45,5 +45,5 @@ func RegisterFeatureGates() {
 
 // DisabledDNSProviderManagement returns true if `DisableDNSProviderManagement` is effective.
 func DisabledDNSProviderManagement() bool {
```

Comment:
Is this function really necessary now? I mean, can't we just replace its occurences with `FeatureGate.Enabled(features.DisableDNSProviderManagement)` ?